src/shared/bus-unit-util.c

```diff
@@ -1415,6 +1418,33 @@ static int bus_append_execute_property(sd_bus_message *m, const char *field, con
                 return 1;
         }
 
+        if (streq(field, "RootHash")) {
+                _cleanup_free_ char *text = NULL;
+                _cleanup_free_ void *roothash_decoded = NULL;
+                size_t roothash_decoded_size = 0;
+
+                if (is_path(eq)) {
+                        if (!path_is_absolute(eq))
```

Comment:
i wouldn't bother with the `is_path` check first. Just go directly for `path_is_absolute()`, and if that fails then call unhexmem(), as you already do... a relative path would then be parsed as hex string, but that's totally OK, and likely immediately result in failure.